layouts/funding_calls/list.html

```diff
@@ -40,154 +37,68 @@ <h4>Potential funding sources</h4>
-                          <div class="modal-header">
-                            <h5 class="modal-title" id="descriptionModalLabel${ind}">${ item.title }</h5>
-                            <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
-                          </div>
-                          <div class="modal-body">
-                            <div class="row mt-1">
-                              <div class="col">
-                                <b>Application deadline:</b> ${item.deadline_time ? long_date + " - " + item.deadline_time : long_date}
-                              </div>
-                            </div>`;
-
-                            if(funder){ 
-                            blockHTML += `
-                            <div class="row mt-2">
-                              <div class="col">
-                                <b>Funder:</b> ${ item.funder }
-                              </div>
-                            </div>`;
-                            } 
-
-                            if(item.description) {
-                              blockHTML += `
-                              <div class="row mt-2">
-                                <div class="col"><b>Brief description:</b><br>
-                                  ${item.description}
-                                </div>
-                              </div>`;
-                            }
-
-                            if(item.applicant){
-                              blockHTML += `
-                              <div class="row mt-2">
-                                <div class="col"><b>Who can apply?</b><br>
-                                  ${item.applicant}
-                                </div>
-                              </div>`;
-                            }
-                            
-
-                            blockHTML += `<div class="row mt-3">
-                              <p>For more information, visit the link below.</p>
-                              <div class="col-md">
-                                <a target="_blank" href="${item.url}"><i class="bi bi-globe"></i> Call webpage</a>
-                              </div>
-                            </div>
-                          </div>
-                          <div class="modal-footer">
-                            <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
-                          </div>
-                        </div>
-                      </div>
-                    </div>
-                  </div>
-                </div>`;
-
-            blockHTML += `
+  // Takes a URL and fetchs all open funding calls with label "Data Centre"
+  async function getDCFundingCalls(url, label="Data Centre") {
+    try {
+      // Fetch DC funding directly from Anubis using API
+      const response = await fetch(url);
+      const allData = await response.json();
```

Comment:
I think it is missing data structure validation. Although i would like to have DS validation all calls but If these APIs are not in our control or not developed then we should definitely validate this.
```suggestion
      const allData = await response.json();
      if (!allData?.calls || !Array.isArray(allData.calls)) {
        return [];
      }
```